### PR TITLE
filter: fix wildcard not working for events

### PR DESCRIPTION
**Pull request by NDStrahilevitz**
## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

```
Author: Nadav Strahilevitz <nadav.strahilevitz@aquasec.com>
Date:   Sun Oct 30 12:27:29 2022 +0000

    filter: fix wildcard not working for events
    
    Preparing events to trace uses the filter's package StringFilter object.
    However this logic doesn't actually need full "filter" logic as such
    a simpler data structure similar to the old string filter is used
    for this logic (sets and events parsing).
```

Fixes: #2306 

## Type of change

- [x] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [ ] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

Do `tracee-ebpf -t e=sched_proc*`

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [ ] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented all functions/methods created explaining what they do.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

File: cmd/tracee-ebpf/flags/filter.go [hunk 1]
```diff
@@ -288,11 +295,10 @@ func PrepareFilter(filtersArr []string) (tracee.Filter, error) {
 	return filter, nil
 }
 
-func prepareEventsToTrace(eventFilter *filters.StringFilter, setFilter *filters.StringFilter, eventsNameToID map[string]events.ID) ([]events.ID, error) {
-	eventFilter.Enable()
```
Comment (yanivagman) [inline cmd/tracee-ebpf/flags/filter.go:292]: I understand that this was not used anywhere, correct?

**Review by yanivagman — COMMENTED** [review]
Not sure I understand how this resolve the wildcard bug. Can you please explain?
Also, it would be nice if you can add a simple test for events with wildcards

File: cmd/tracee-ebpf/flags/filter.go [hunk 2]
```diff
@@ -288,11 +295,10 @@ func PrepareFilter(filtersArr []string) (tracee.Filter, error) {
...
-	eventsToTrace := eventFilter.Equal()
-	excludeEvents := eventFilter.NotEqual()
-	setsToTrace := setFilter.Equal()
+func prepareEventsToTrace(eventFilter cliFilter, setFilter cliFilter, eventsNameToID map[string]events.ID) ([]events.ID, error) {
+	eventsToTrace := eventFilter.Equal
+	excludeEvents := eventFilter.NotEqual
+	setsToTrace := setFilter.NotEqual
```
[the remark this answers is not in the document]
Comment (NDStrahilevitz) [reply cmd/tracee-ebpf/flags/filter.go:301]: I believe it wasn't correct before, see: https://github.com/aquasecurity/tracee/blob/d5320edaca9891eb709081e1998f8cc2715cce46/cmd/tracee-ebpf/flags/filter.go#L322 in a previous commit before the previous filters PR.

File: cmd/tracee-ebpf/flags/filter.go (hunk 1 above)
Comment (NDStrahilevitz) [reply cmd/tracee-ebpf/flags/filter.go:292]: Correct.

File: cmd/tracee-ebpf/flags/filter.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (NDStrahilevitz) [reply cmd/tracee-ebpf/flags/filter.go:301]: Aah you meant the setsToTrace, you're right it's a mistake, i'll fix it.

**Review by yanivagman — APPROVED** [review]
LGTM

**Pull request merged**